### PR TITLE
Changes to the submitter

### PR DESCRIPTION
**Pull request by anmalara**
Proof of code, simplified, add functionalities

### CODE REVIEW

File: Submitter/submit.py [hunk 1]
```diff
@@ -29,35 +29,26 @@ def main():
...
-    forceadd = args.forceadd
-    notreeadd = args.notreeadd
-    hadd = args.hadd
-    clean = args.clean
-    ncores = args.ncores
-    nargs = sum([1 for x in vars(args) if vars(args)[x] is True])
 
     ROOT.gErrorIgnoreLevel = kError
-
+    RunLocal  = args.ncores is not None
```
Comment (reimersa) [inline Submitter/submit.py:34]: Please use all lower-case variable names. `RunLocal` is also the name of a member function and therefore particularly dangerous

File: Submitter/submit.py [hunk 2]
```diff
@@ -29,35 +29,26 @@ def main():
...
+    isToAdd   = args.add or args.forceadd or args.notreeadd
```
Comment (reimersa) [inline Submitter/submit.py:35]: Please use all lower-case variable names like is_to_add

**Review by reimersa — CHANGES_REQUESTED** [review]

File: Submitter/submit.py [hunk 3]
```diff
@@ -30,7 +30,7 @@ def main():
     if not xmlfilename.endswith('.xml'):
         raise ValueError(red('The name of the xml-file does not end with \'.xml\'. I don\'t believe you...'))
 
-    ROOT.gErrorIgnoreLevel = kError
+    # ROOT.gErrorIgnoreLevel = kError
```
Comment (reimersa) [inline Submitter/submit.py:33]: Did you check if this changes any behavior? If not, please simply delete the line

**Review by reimersa — CHANGES_REQUESTED** [review]

**Review by reimersa — APPROVED** [review]
Nice addition!

**Review by reimersa — CHANGES_REQUESTED** [review]
Oops, did not mean to approve

File: Submitter/submit.py (hunk 3 above)
Comment (anmalara) [reply Submitter/submit.py:33]: I tested with -c, -cs, -ds. Didn't see a difference. Not sure what happens if it crashes though

Comment (reimersa) [reply Submitter/submit.py:33]: If it crashes, this line shouldn't protect us either... I'd just remove it

File: Submitter/XMLInfo.py [hunk 4]
```diff
@@ -10,9 +10,12 @@ def __init__(self, xmlfilename):
         self.xmlfilename = xmlfilename
         command = 'xmllint --noent --dtdattr %s' % (xmlfilename)
         xmlstring = StringIO.StringIO(subprocess.check_output(command.split(' ')))
```
Comment (reimersa) [inline Submitter/XMLInfo.py:12]: please remove this line, it is duplicated in new line 15 now

File: Submitter/XMLInfo.py [hunk 5]
```diff
@@ -10,9 +10,12 @@ def __init__(self, xmlfilename):
...
+        self.ConfigInfo = '\n'.join(xmlstring.read().split('\n')[0:2])+']>\n'
+        self.ConfigName = self.ConfigInfo[self.ConfigInfo[:self.ConfigInfo.find('.dtd')+len('.dtd')].rfind('"')+1:self.ConfigInfo.find('.dtd')+len('.dtd')]
```
Comment (reimersa) [inline Submitter/XMLInfo.py:14]: please make member variable names all lowercase, starting with a capital letter is reserved for main member functions called from outside the class

**Review by reimersa — CHANGES_REQUESTED** [review]

**Review by reimersa — COMMENTED** [review]
Thanks, this file is good now

File: Submitter/XMLInfo.py (hunk 4 above)
Comment (anmalara) [reply Submitter/XMLInfo.py:12]: this is needed because after line 13 the read function empty the StringIO content. Didn't find a better option here

Comment (reimersa) [reply Submitter/XMLInfo.py:12]: ugh, I see. Fine!

File: Submitter/XMLInfo.py [hunk 6]
```diff
@@ -10,12 +10,12 @@ def __init__(self, xmlfilename):
         self.xmlfilename = xmlfilename
         command = 'xmllint --noent --dtdattr %s' % (xmlfilename)
         xmlstring = StringIO.StringIO(subprocess.check_output(command.split(' ')))
-        self.ConfigInfo = '\n'.join(xmlstring.read().split('\n')[0:2])+']>\n'
-        self.ConfigName = self.ConfigInfo[self.ConfigInfo[:self.ConfigInfo.find('.dtd')+len('.dtd')].rfind('"')+1:self.ConfigInfo.find('.dtd')+len('.dtd')]
+        self.config_info = '\n'.join(xmlstring.read().split('\n')[0:2])+']>\n'
+        self.config_name = self.config_info[self.config_info[:self.config_info.find('.dtd')+len('.dtd')].rfind('"')+1:self.config_info.find('.dtd')+len('.dtd')]
```
Comment (reimersa) [inline Submitter/XMLInfo.py:14]: I trust you on this one, the line is too long for me to understand

Comment (anmalara) [reply Submitter/XMLInfo.py:14]: just for reference: it selects a string, then find the 'dtd' part and select all the info from the '.dtd' back to the first quotation symbol. It's the current structure of the XML file. I think it's rather safe like this (although ugly)

**Comment by reimersa** [comment]
Will skip CI, this won't affect compliation.

**Pull request merged**